app/controllers/application_controller.rb

```diff
@@ -24,25 +24,22 @@ class ApplicationController < Sinatra::Base
     @bin = Bin.new(params[:bin])
 
     if @bin.save
-      content_type :json
-      { id: @bin.id }.to_json
+      #content_type :json
+      #{ id: @bin.id }.to_json
+      erb :create
```

Comment:
We need to return a json here. I think we need an ajax-call for sending the encrypted bin via POST and get an ID in return. In the index-view we need javascript-kungfu so that we can modify the output with the id we returned by the ajax call and the SECRET we generated with javascript

Reply:
Hmm... so we could return json and let JS redirect to something like /created/:random_id so we don't have to render the success-message with js

Reply:
I changed it back to json and added a test for the json response:

https://github.com/aha-oida/aha-secret/pull/21/files#diff-459d782ca686bf3fc2ea366d22bb5b851dad7008b5e95fcd413062e516edd92fR20